File: src/main/java/org/example/bookfd/controllers/MenuController.java
package org.example.bookfd.controllers;

import org.springframework.security.core.Authentication;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

import java.util.Map;

@Controller
public class MenuController {

    @GetMapping("/")
    public String start(Map<String, Object> model, Authentication authentication) {
        model.put("userstatus",authentication.getAuthorities());
        return "StartPage";
    }

    @GetMapping("/menu")
    public String menu(Map<String, Object> model){

        return "menu";
    }

}

File: src/main/java/org/example/bookfd/controllers/MenuController.kt
package org.example.bookfd.controllers

import org.springframework.security.core.Authentication
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

@Controller
class MenuController {

    @GetMapping("/")
    fun start(model: MutableMap<String?, Any?>, authentication: Authentication): String {
        model["userstatus"] = authentication.authorities
        return "menu/TopMenu"
    }

    @GetMapping("/menu")
    fun menu(): String {
        return "menu"
    }
}